gno.land/pkg/gnoland/app.go

```diff
@@ -20,34 +22,68 @@ import (
 	"github.com/gnolang/gno/tm2/pkg/store/iavl"
 )
 
+type CustomAppConfig struct {
+	DB dbm.DB
+	// `gnoRootDir` should point to the local location of the gno repository.
+	// It serves as the gno equivalent of GOROOT.
+	GnoRootDir            string
+	SkipFailingGenesisTxs bool
+	Logger                log.Logger
+	MaxCycles             int64
+}
+
+func (c *CustomAppConfig) ApplyDefault() {
+	if c.Logger == nil {
+		c.Logger = log.NewNopLogger()
+	}
+
+	if c.DB == nil {
+		c.DB = dbm.NewMemDB()
+	}
+
+	if c.GnoRootDir == "" {
+		c.GnoRootDir = guessGnoRootDir()
+	}
+}
+
+func (c *CustomAppConfig) validate() error {
+	if c.Logger == nil {
+		return fmt.Errorf("no logger provided")
+	}
+
+	if c.DB == nil {
+		return fmt.Errorf("no db provided")
+	}
+
+	return nil
+}
+
 // NewApp creates the GnoLand application.
-func NewApp(rootDir string, skipFailingGenesisTxs bool, logger log.Logger, maxCycles int64) (abci.Application, error) {
-	// Get main DB.
-	db, err := dbm.NewDB("gnolang", dbm.GoLevelDBBackend, filepath.Join(rootDir, "data"))
-	if err != nil {
-		return nil, fmt.Errorf("error initializing database %q using path %q: %w", dbm.GoLevelDBBackend, rootDir, err)
+func NewCustomApp(cfg CustomAppConfig) (abci.Application, error) {
```

Comment:
Instead of calling it a "custom app", why don't we make this one `NewAppWithOptions` and the config `AppOptions`?

Reply:
im not good at naming method, it's definitely looks better, I will update this.